### ISSUE TITLE
Help running tests

### ISSUE THREAD

**Issue by hdoupe** [open-source-economics/OG-USA#307]
I’m having some trouble running py.test for OG-USA.  Which directly should I run it from and which command should I use?

I keep getting this error: 

```
(ogusatemp) HDoupe-MacBook-Pro:ogusa henrydoupe$ py.test
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 2.7.13, pytest-3.1.2, py-1.4.34, pluggy-0.4.0
rootdir: /Users/henrydoupe/Documents/OG-USA, inifile: setup.cfg
collected 6 items / 1 errors 

=================================================================================================================================== ERRORS ===================================================================================================================================
_____________________________________________________________________________________________________________ ERROR collecting Python/ogusa/tests/test_basic.py ______________________________________________________________________________________________________________
ImportError while importing test module '/Users/henrydoupe/Documents/OG-USA/Python/ogusa/tests/test_basic.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_basic.py:8: in <module>
    from ogusa.utils import comp_array
E   ImportError: No module named ogusa.utils
============================================================================================================================== warnings summary ==============================================================================================================================
None
  [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.

-- Docs: http://doc.pytest.org/en/latest/warnings.html
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================================================================================== 1 warnings, 1 error in 1.73 seconds =====================================================================================================================

```

where ogusatemp is a temporary anaconda environment I made from PR #306.

**Comment by jdebacker** [comment]
@hdoupe I'm not sure this `py.test` is working at all of OG-USA.  I'm using the environment in OG-USA:master and I'm getting the same error.

I think it would be safe to just start from scratch with the local testing.  You might  look to what is done in Travis CI for OG-USA to see what testing has been working.

**Comment by hdoupe** [comment]
@jdebacker said 

> I think it would be safe to just start from scratch with the local testing. You might look to what is done in Travis CI for OG-USA to see what testing has been working.

Yeah, I agree.  There's a problem with how py.test sets up the paths.  I made a fake test file to see what was in `sys.path`.

```
import os
import sys

def test_not():
    with open('/Users/henrydoupe/Documents/OG-USA/Python/ogusa/tests/test_status.txt', 'w') as w:
        w.write('\n'.join(os.sys.path))
```

Then, I ran `py.test ogusa/tests/test_fake.py` from the `Python` directory.  The contents of `sys.path` are 

```
/Users/henrydoupe/Documents/OG-USA/Python/ogusa/tests
/anaconda/envs/ogusatemp/bin
/anaconda/envs/ogusatemp/lib/python27.zip
/anaconda/envs/ogusatemp/lib/python2.7
/anaconda/envs/ogusatemp/lib/python2.7/plat-darwin
/anaconda/envs/ogusatemp/lib/python2.7/plat-mac
/anaconda/envs/ogusatemp/lib/python2.7/plat-mac/lib-scriptpackages
/anaconda/envs/ogusatemp/lib/python2.7/lib-tk
/anaconda/envs/ogusatemp/lib/python2.7/lib-old
/anaconda/envs/ogusatemp/lib/python2.7/lib-dynload
/anaconda/envs/ogusatemp/lib/python2.7/site-packages
/anaconda/envs/ogusatemp/lib/python2.7/site-packages/setuptools-27.2.0-py2.7.egg
```

Then performing the same fake test in taxcalc, I got the following result

```
/Users/henrydoupe/Documents/Tax-Calculator/taxcalc/tests
/Users/henrydoupe/Documents/Tax-Calculator
/anaconda/envs/taxcalc-dev/bin
/anaconda/envs/taxcalc-dev/lib/python36.zip
/anaconda/envs/taxcalc-dev/lib/python3.6
/anaconda/envs/taxcalc-dev/lib/python3.6/lib-dynload
/anaconda/envs/taxcalc-dev/lib/python3.6/site-packages
/anaconda/envs/taxcalc-dev/lib/python3.6/site-packages/pytest_cache-1.0-py3.6.egg
/anaconda/envs/taxcalc-dev/lib/python3.6/site-packages/pytest_pep8-1.0.6-py3.6.egg
/anaconda/envs/taxcalc-dev/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg
```

So this path is missing in the ogusa test path`/Users/henrydoupe/Documents/OG-USA/Python/` which corresponds to this path in taxcalc: `/Users/henrydoupe/Documents/Tax-Calculator`.  I'm trying to figure out how to solve this now.

In the meantime, you can run `python -m pytest ogusa/tests` and it works.

**Comment by PeterDSteinberg** [comment]
@hdoupe @jdebacker This issue appears related: https://github.com/open-source-economics/OG-USA/issues/290 I encountered the same issue you are talking about (`ogusa.utils` import problem) after doing `python setup.py develop` but did not encounter that after doing `python setup.py install` .  I can look into this more before the meeting.

**Comment by MattHJensen** [comment]
@PeterDSteinberg, let @hdoupe pursue this one. You've got a full plate.

**Comment by hdoupe** [comment]
@PeterDSteinberg I ran `python setup.py install` and it works now.  But I'm still not sure why I have to do that for ogusa but not for taxcalc.  I'll look into it some more.  

Thanks for you help!

**Comment by PeterDSteinberg** [comment]
No problem @hdoupe - I have been meaning to fix that issue for a while ( #290 ).  I'll close this issue as a duplicate essentially of #290 .  I'll take a look at the setup.py before tomorrow afternoon

**Comment by PeterDSteinberg** [comment]
@hdoupe - I didn't read clearly the whole thread inclusive of @MattHJensen 's comment that you'll be handling this setup.py bug ( #290 ) - let me know if you have any trouble.
